http.kafka.sync/test.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# GIVEN
+ZILLA_PORT="7114"
+KAFKA_PORT="9092"
+ITEM_ID="5cf7a1d5-3772-49ef-86e7-ba6f2c7d7d07"
+GREETING="Hello, World!"
+GREETING_DATE="Hello, World! $(date)"
+EXPECTED="{\"greeting\":\"$GREETING_DATE\"}"
+
+echo \# Testing http.kafka.sync
+echo ZILLA_PORT=$ZILLA_PORT
+echo KAFKA_PORT=$KAFKA_PORT
+echo ITEM_ID=$ITEM_ID
+echo GREETING=$GREETING
+echo GREETING_DATE=$GREETING_DATE
+echo EXPECTED=$EXPECTED
+echo
+
+# WHEN
+# send request to zilla
+timeout 300 curl -vs \
+       -X "PUT" http://localhost:$ZILLA_PORT/items/$ITEM_ID \
+       -H "Idempotency-Key: 1" \
+       -H "Content-Type: application/json" \
+       -d "{\"greeting\":\"$GREETING\"}" | tee .testoutput &
+
+# fetch correlation id from kafka with kcat; retry until ready
+for i in $(seq 1 20); do
+  CORRELATION_ID=$(timeout 10 kcat -C -b localhost:$KAFKA_PORT -t items-requests -J -u | jq -r '.headers | index("zilla:correlation-id") as $index | .[$index + 1]')
+  if [[ ! -z "$CORRELATION_ID" ]]; then
+    break
+  fi
+done
+echo CORRELATION_ID=$CORRELATION_ID
+if [ -z "$CORRELATION_ID" ]; then
+  echo ❌
+  exit 1
+fi
```

Comment:
Is it necessary to repeat the failure output here when the test fails early?
In the general case, where it is possible to fail early in N different ways, this would be repeated N times.

For example, if each part is a function, then the execution invokes each function in turn, skipping over functions after early failure, falling through to output success or failure once.

Reply:
Actually my preference is to repeat those two lines (echo x; exit 1) whenever it makes sense to exit prematurely as opposed to introduce complex control flow with functions and such. This way we can keep a linear execution flow with the only exception of exiting early in case of fatal errors. I hope it's OK this way.